File: Trik.java
import java.util.Scanner;

public class Trik {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		String input = sc.nextLine();
		
		int loc = 1;
		
		for(int i = 0; i < input.length(); i++){
			if(loc == 1 && input.charAt(i) == 'A') {
				loc = 2;
			} else if(loc == 1 && input.charAt(i) == 'C') {
				loc = 3;
			} else if(loc == 2 && input.charAt(i) == 'A') {
				loc = 1;
			} else if(loc == 2 && input.charAt(i) == 'B') {
				loc = 3;
			} else if(loc == 3 && input.charAt(i) == 'C') {
				loc = 1;
			} else if(loc == 3 && input.charAt(i) == 'B') {
				loc = 2;
			}
		}
		
		System.out.println(loc);
		
	}

}

File: Kattis-Problems/Python-Solutions/Trik.py
a = input()
loc = 1

for i in range(len(a)):
    move = a[i:i+1]
    if(move == 'A' and loc == 1):
        loc = 2
    elif(move =='A' and loc == 2):
        loc = 1
    elif(move == 'B' and loc == 2):
        loc = 3
    elif(move == 'B' and loc == 3):
        loc = 2
    elif(move == 'C' and loc == 1):
        loc = 3
    elif(move == 'C' and loc == 3):
        loc = 1
    else:
        continue

print(loc)